### PR TITLE
Unit test api endpoints

### PR DESCRIPTION
**Pull request by dwarukira**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -101,3 +101,4 @@ ENV/
 .mypy_cache/
 .vscode/
 UI/images/tick.svg
+*.swp
```
Comment (kwahalf) [inline .gitignore]: add a new line after every file

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+python: 
+  - "3.6"
+
+cache: pip
+install: 
+  - pip install -r requirements.txt
+
+
+script: 
+  - nosetests
```
Comment (kwahalf) [inline .travis.yml]: same here

File: app/apiv1/views.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    username = data['username']
+    password = data['password']
+
+    for user in users:
+        if username == user.username:
+            if user.check_password(password):
+                return jsonify({'status':200 , 'token':user.get_token()})
+            return jsonify({'status':401, 'message':'invalid password'}), 401
+
+    return jsonify({'status':401, 'message':'invalid username and password'}), 401
```
Comment (kwahalf) [inline app/apiv1/views.py]: same here new line

File: app/models.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        
+    def to_dict(self):
+        user_dict = {
+            'username':self.username,
+            'phone':self.phone,
+            'email':self.email,
+            'is_admin':self.is_admin
+        }
+
+        return user_dict
```
Comment (kwahalf) [inline app/models.py]: new line

File: test/test_config.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import unittest
```
Comment (kwahalf) [inline test/test_config.py]: what does this file do?

**Comment by kwahalf** [comment]
and Update the pull request to be to develop not master

File: app/apiv1/auth_views.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        phone = request.json['phone']
+        password = request.json['password']
+        role = request.json.get('role', None)
+        if role == None:
+            role =  1
+        user = User(username, phone, password, role=role, email=email)
+        
+        for u in Store.users:
+            if username == u.username:
+                return {'status':409 , 'message':'username is taken'}, 409
```
Comment (kwahalf) [inline app/apiv1/auth_views.py]: use correct status code

File: app/apiv1/auth_views.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        Store.users.append(user)
+
+        return {'status':201, 'message':'User created', 'token':user.get_token()}, 201
+
+class Login(Resource):
+    ''' Register APi resource'''
+    
+    def post(self):
+        username  = request.json.get('username', None)
+        # email = request.json.get('email', None)
```
Comment (kwahalf) [inline app/apiv1/auth_views.py]: do not commit commented out code

File: run.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+''' run the app '''
+
+import os
+
+from app import new_app
+
+mode = os.environ.get('MODE') if os.environ.get('MODE') else "development"
+
+app = new_app(mode)
```
Comment (kwahalf) [inline run.py]: how will this app run?

File: test/test_app_api.py [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+    
+    def tearDown(self):
+        pass
+    
+    def test_root_view(self):
+        ''' test root view'''
+        res = self.app.get("/")
+        self.assertEqual(res.status_code, 404)
+
+    def test_signup(self):
```
Comment (kwahalf) [inline test/test_app_api.py]: test one thing at a time ie  test someone can sign up successfully, then test that if someone can't sign up twice

File: test/test_meals_api.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+        self.assertEqual(res.status_code, 200)
+
+        res = self.app.get("/api/v1/meal/1",
+                           headers=dict(Authorization="Bearer " + self.token))
+
+        self.assertEqual(res.status_code, 404)
+    
+
+
```
Comment (kwahalf) [inline test/test_meals_api.py]: new line

File: app/models.py [hunk 11]
```diff
@@ -0,0 +1,157 @@
+from werkzeug.security import generate_password_hash , check_password_hash
+import uuid
+from datetime import datetime
+from flask import current_app
+from flask_jwt_extended import create_access_token
+from flask_restful import url_for
+
+
```
Comment (kwahalf) [inline app/models.py]: make sure your code follows pep8

File: app/apiv1/auth_views.py (hunk 6 above)
Comment (dwarukira) [reply app/apiv1/auth_views.py]: 409 is conflict

**Pull request merged**